worker/uniter/pebblenotices.go

```diff
@@ -163,7 +163,12 @@ func (n *pebbleNoticer) processNotice(containerName string, notice *client.Notic
 		// anything else, so cannot miss the change entirely.
 		chg, err := pebbleClient.Change(notice.Key)
 		if err != nil {
```

Comment:
I think we should only recover from the not found error (i.e., 404) since the API call can fail for other reasons. However, I briefly checked the Pebble client in Juju, and it doesn’t provide that level of detail in errors. So, if we want to detect only not found errors, we will need to modify that as well...

Reply:
Yes, that's a valid point. Although the consequence would be that some other error would cause a false-positive check event being generated. That doesn't seem *too* bad, since when you get the event you are meant to ask Pebble whether the check is currently up/down *right then*, as the Juju event may be old news - it's saying it happened, not what the state is right now. So a well-written charm would properly ignore a false positive, other than perhaps keeping track of how many times a check had gone to down.

I definitely think it's better for Juju to incorrectly emit a check event than to incorrectly skip a check event.

Reply:
Ah, that's fair. The pebble client does actually have [`client.Error`](https://pkg.go.dev/github.com/canonical/pebble@v1.18.0/client#Error) and you can check if the `StatusCode` field is 404. It's a reasonable suggestion; I'll make that change on Monday.

Reply:
> I definitely think it's better for Juju to incorrectly emit a check event than to incorrectly skip a check event.

Ah, sorry @tonyandrewmeyer, only saw your comment after I'd posted mine. Do you mean you think it's better to leave it as is (very tiny chance of incorrectly emitting)?

Reply:
No, I was meaning it's better to have it as it is now in the PR than the earlier version that would skip emitting the event. Only assuming it's pruned on 404 is better than both, and if that can be done without Pebble changes that's definitely best.

Reply:
Fixed.